greenlet.c

```diff
@@ -1044,11 +1050,11 @@ static void green_dealloc(PyGreenlet* self)
 		 * it would cause a recursive call.
 		 */
 		assert(Py_REFCNT(self) > 0);
-		if (--Py_REFCNT(self) != 0) {
+		if (--((PyObject*)(self))->ob_refcnt != 0) {
```

Comment:
Just noticed, direct field access like this is incorrect. The whole reason for new macros is for this field to eventually go away. This should probably be something like:

```
Py_ssize_t refcnt = Py_REFCNT(self) - 1;
Py_SET_REFCNT(self, refcnt);
if (refcnt != 0) {
    ...
}
```

Reply:
Yeah, but both ways seemed ugly for me. But i re-updated it according to your request